File: 35-search-insert-position/search-insert-position.go
func searchInsert(nums []int, target int) int {
    if nums[0] > target {
        return 0;
    }

    for i:= 0; i < len(nums); i++ {
        if nums[i] >= target {
            return i;
        }
    }

    return len(nums);
}

File: 35-search-insert-position/search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int left = 0, right = nums.length - 1;

        while (left <= right) {
            int mid = left + (right - left) / 2;

            if (nums[mid] == target) {
                return mid; // Found exact match
            } else if (nums[mid] < target) {
                left = mid + 1; // Move right
            } else {
                right = mid - 1; // Move left
            }
        }

        return left; // left is the correct insertion point
    }
}